distribution/sender/routes/debezium-route.xml

```diff
@@ -3,88 +3,101 @@
         <from uri="debezium-mysql:extract?databaseServerId={{debezium.db.serverId}}&amp;databaseServerName={{debezium.db.serverName}}&amp;databaseHostname={{openmrs.db.host}}&amp;databasePort={{openmrs.db.port}}&amp;databaseUser={{debezium.db.user}}&amp;databasePassword={{debezium.db.password}}&amp;databaseWhitelist={{openmrs.db.name}}&amp;offsetStorageFileName={{debezium.offsetFilename}}&amp;databaseHistoryFileFilename={{debezium.historyFilename}}&amp;tableWhitelist={{debezium.tablesToSync}}&amp;offsetFlushIntervalMs=0&amp;snapshotMode={{debezium.snapshotMode}}&amp;snapshotFetchSize=1000&amp;snapshotLockingMode=extended&amp;includeSchemaChanges=false" />
 
         <log message=":" loggingLevel="DEBUG" />
-        <log message="Received DB event: Operation=${headers.CamelDebeziumOperation}, Body=${body}, Metadata=${headers.CamelDebeziumSourceMetadata}" />
-        <setProperty name="event">
-            <spel>#{new org.openmrs.eip.component.entity.Event()}</spel>
+        <log message="Received DB event: Operation=${headers.CamelDebeziumOperation}, Body=${body}, Metadata=${headers.CamelDebeziumSourceMetadata}" loggingLevel="INFO"/>
+
+        <setProperty name="debeziumEvent">
+            <spel>#{new org.openmrs.eip.app.management.entity.DebeziumEvent()}</spel>
         </setProperty>
+        
         <script>
             <spel>
                 <!-- TODO Take care of situation where a table has a composite FK -->
-                #{getProperty('event').setPrimaryKeyId(request.headers.CamelDebeziumKey.get(request.headers.CamelDebeziumKey.schema().fields().get(0)))}
-                #{getProperty('event').setTableName(request.headers.CamelDebeziumSourceMetadata.get('table'))}
-                #{getProperty('event').setOperation(request.headers.CamelDebeziumOperation)}
-                #{getProperty('event').setSnapshot(request.headers.CamelDebeziumSourceMetadata.get('snapshot') != 'false')}
+                #{getProperty('debeziumEvent').event.setPrimaryKeyId(request.headers.CamelDebeziumKey.get(request.headers.CamelDebeziumKey.schema().fields().get(0)))}
+                #{getProperty('debeziumEvent').event.setTableName(request.headers.CamelDebeziumSourceMetadata.get('table'))}
+                #{getProperty('debeziumEvent').event.setOperation(request.headers.CamelDebeziumOperation)}
+                #{getProperty('debeziumEvent').event.setSnapshot(request.headers.CamelDebeziumSourceMetadata.get('snapshot') != 'false')}
+                #{getProperty('debeziumEvent').setDateCreated(new java.util.Date())}
+      			#{getProperty('debeziumEvent').setEventTimeStamp(request.headers.CamelDebeziumTimestamp)}
             </spel>
         </script>
+
         <setProperty name="is-subclass">
```

Comment:
This logic should not be in this route

Reply:
Again..
If you put this logic on processor-route I think we will have troubles. Because this route receves records from reader-route and  I can't figure out how will we find the UUID after the record have been removed. Unless you mean to create a new route and put this logic and then call this new route from debezium-route